### ISSUE TITLE
Remove support for sect571k1 as it is no longer supported by Java-17

### ISSUE THREAD

**Issue by DarshitChanpura** [opensearch-project/security#1620]
#### A known JDK bug [JDK-8251547](https://bugs.openjdk.java.net/browse/JDK-8251547) is causing CI build to fail for Java-17 with the following error:

```
com.amazon.dlic.auth.http.jwt.HTTPJwtAuthenticatorTest > testES512 FAILED
    java.security.ProviderException: Curve not supported:  sect571k1 [NIST K-571] (1.3.132.0.38)
        at jdk.crypto.ec/sun.security.ec.ECKeyPairGenerator.generateKeyPair(ECKeyPairGenerator.java:157)
        at java.base/java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:727)
        at com.amazon.dlic.auth.http.jwt.HTTPJwtAuthenticatorTest.testES512(HTTPJwtAuthenticatorTest.java:489)
```

> Note: A temporary fix has been implement in #1609 to ignore the broken test. 

This issue needs to be triaged to align with one of the two options: 
1. Remove elliptic curve sect571k1 support completely
2. Add support for this curve when being built using JDK 17

**Comment by dblock** [comment]
This seems to say that the test fails, but is the problem JDK17 support in the security plugin, or is it just a test that needs to be rewritten?  It seems that `sect571k1` is no longer supported on JDK17. What does this mean to users? What do we need to document? What will be broken for those who use JDK17?

**Comment by davidlago** [comment]
@DarshitChanpura could you please see if there are any alternative providers for this EC that we could add instead? I'm not 100% sure the reason for removal was security concerns but rather lack of use/outdated.

**Comment by peternied** [comment]
After https://github.com/opensearch-project/security/pull/1661 gets merged can we see if we can re-enable this test?

**Comment by davidlago** [comment]
We will use the opportunity of the breaking change in 2.0 to drop support for this EC. If people still need this we can open an issue down the road and study options for alternative providers of these.

**Comment by DarshitChanpura** [comment]
> After #1661 gets merged can we see if we can re-enable this test?

Even after #1661 's merge, the test still fail with the same issue.

**Comment by peternied** [comment]
Going to make the code change to make sure this is resolved

**Comment by DarshitChanpura** [comment]
Once the PR: [#1771 ](https://github.com/opensearch-project/security/pull/1711) is merged we can close: #1670
